--- a/gcc/ginclude/float.h
+++ b/gcc/ginclude/float.h
@@ -629,3 +631,9 @@
 #endif
 
 #endif /* _FLOAT_H___ */
+
+#ifdef _LIBCPP_CFLOAT
+#if __has_include_next(<float.h>)
+#  include_next <float.h>
+#endif
+#endif
--- a/gcc/ginclude/stddef.h
+++ b/gcc/ginclude/stddef.h
@@ -461,3 +461,9 @@
 
 #endif /* !_STDDEF_H && !_STDDEF_H_ && !_ANSI_STDDEF_H && !__STDDEF_H__
 	  || __need_XXX was not defined before */
+
+#ifdef _LIBCPP_CSTDDEF
+#if __has_include_next(<stddef.h>)
+#  include_next <stddef.h>
+#endif
+#endif
